### ISSUE TITLE
`in` operator does not check type of element

### ISSUE THREAD

**Issue by Wilfred** [python/mypy#4645]
Given the following code:

```python
from typing import List

def contains_foo(list_of_ints):
    # type: (List[int]) -> bool
    return 'foo' in list_of_ints
```

`mypy --strict --py2 example.py` does not complain that I'm looking for strings in a list of ints.

I don't think there are duplicate issues, but `in` is a tricky thing to search for. [This comment in #4071](https://github.com/python/mypy/issues/4071#issuecomment-335029437) is possibly relevant, talking about narrowing.

**Comment by gvanrossum** [comment]
The problem is that quite often people use this behavior of `in` (and `==`) in working code, depending on the fact that e.g. `None` or some other value is not in a given list (or not equal to a given value). So "fixing" this would cause a lot of pain for people who have painstakingly annotated legacy codebases. (Like my employer.)

**Comment by JukkaL** [comment]
This is a duplicate of #1749.
